### PR TITLE
manage new hook name displayProductAdditionalInfo

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | master
| Description?  | manage new hook name displayProductAdditionalInfo introduced in 1.7.1, add missing class for flex products list, better management of product list to avoid getting too many products, and integrate minor modifications
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/9916
| How to test?  | See forge ticket

### CODE REVIEW

File: ps_viewedproduct.php [hunk 1]
```diff
@@ -37,12 +37,14 @@
...
 {
     private $templateFile;
+    private $currentProductId;
 
     public function __construct()
     {
         $this->name = 'ps_viewedproduct';
         $this->author = 'PrestaShop';
-        $this->version = '1.0.1';
+        $this->version = '1.0.2';
```
Comment (Quetzacoalt91) [inline ps_viewedproduct.php]: As you modify the features of the module, a version v1.1.0 seems more suitable.

File: ps_viewedproduct.php [hunk 2]
```diff
@@ -54,36 +56,36 @@ public function __construct()
...
     }
 
     public function install()
     {
         return parent::install()
             && Configuration::updateValue('PRODUCTS_VIEWED_NBR', 8)
             && $this->registerHook('displayFooterProduct')
-            && $this->registerHook('displayProductButtons')
+            && $this->registerHook('displayProductButtons') //Former version of displayProductAdditionalInfo
+            && $this->registerHook('displayProductAdditionalInfo')
```
Comment (Quetzacoalt91) [inline ps_viewedproduct.php]: This method is only called for merchant installing the module.

For the others, you need to handle the upgrade which register the hook on the new version: https://devdocs.prestashop.com/1.7/modules/creation/enabling-auto-update/

File: ps_viewedproduct.php (hunk 1 above)
Comment (jolelievre) [reply ps_viewedproduct.php]: well it is more an incompatibility with the latest PrestaShop version, do you consider this a new feature?

Comment (Quetzacoalt91) [reply ps_viewedproduct.php]: From a module point of view, you update its features. :)

Comment (jolelievre) [reply ps_viewedproduct.php]: Ok I will change it!

File: ps_viewedproduct.php (hunk 2 above)
Comment (jolelievre) [reply ps_viewedproduct.php]: done ^^

File: ps_viewedproduct.php (hunk 1 above)
Comment (jolelievre) [reply ps_viewedproduct.php]: done!

File: ps_viewedproduct.php [hunk 3]
```diff
@@ -135,7 +149,8 @@ public function renderForm()
         $helper->show_toolbar = false;
         $helper->table =  $this->table;
         $helper->default_form_language = $lang->id;
-        $helper->allow_employee_form_lang = Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') ? Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') : 0;
+        $configFormLang = Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG');
+        $helper->allow_employee_form_lang = $configFormLang = $configFormLang ? $configFormLang : 0;
```
Comment (Quetzacoalt91) [inline ps_viewedproduct.php]: Affectation to `$configFormLang` by mistake?

File: ps_viewedproduct.php [hunk 4]
```diff
@@ -166,10 +181,11 @@ public function getCacheId($name = null)
         return parent::getCacheId('ps_viewedproduct|' . $key);
     }
 
-    public function renderWidget($hookName = null, array $configuration = [])
+    public function renderWidget($hookName = null, array $configuration = array())
     {
-        if ('displayProductButtons' === $hookName) {
-            $this->addViewedProduct($configuration['product']['id_product']);
+        $this->currentProductId = $configuration['product']['id_product'];
```
Comment (Quetzacoalt91) [inline ps_viewedproduct.php]: This information may not be available anywhere, that's why it was inside the `if()` statement

**Comment by jolelievre** [comment]
@Quetzacoalt91 I switched this one to QA but all your requested changes should be done ;)

**Comment by marionf** [comment]
@jolelievre 

I have an exception if I try to transplant the module on the left or right column

capture d ecran_281

**Comment by jolelievre** [comment]
@marionf thanks, I didn't know this feature.. This is fixed!

**Comment by marionf** [comment]
All good, QA approved :+1: 
thank you @jolelievre

**Comment by PierreRambaud** [comment]
@Quetzacoalt91 Should we target dev instead of master branch?

**Comment by Quetzacoalt91** [comment]
Yep, PRs must target the `dev` branch.

**Comment by gudmund1** [comment]
Hi, is there an updated version of this module with these fixes applied? I have searched addons but I can't find the module anywhere...

**Comment by jolelievre** [comment]
This is odd, I can see it in the modules list:
capture d ecran 2018-11-07 a 10 35 03

Did you, by any chance, disable the addons update when there were problems on the platform a few weeks ago? There was a woraround about switching `Tools::is_addons_up` to false
(just an idea which could explain your issue)

**Comment by gudmund1** [comment]
Aaah, yes that is probably the problem. Is it safe to uninstall this module now? Thanks for the help!!
________________________________

**Pull request merged**